### PR TITLE
Correct some pkgconfig file errors

### PR DESCRIPTION
**Pull request by nhorman**
@eli-schwartz noted a few shortocomings in the pkgconfig file added in PR 334:
* Failure to honor CMAKE_INSTALL_LIBDIR/CMAKE_INSTALL_INCLUDEDIR
* duplication of ldflags in the Libs.private section

This PR fixes those, by:
* Using cmake_path to construct the full lib/include paths prior to doing macro expansion on the pc.in file
* removing the duplication library flags in the private section for static linking

Signed-off-by: Neil Horman <nhorman@gmail.com>

### CODE REVIEW

File: libteec/libteec.pc.in [hunk 1]
```diff
@@ -13,8 +13,6 @@ Name: @PROJECT_NAME@
 Description: @CMAKE_PROJECT_DESCRIPTION@
 URL: @CMAKE_PROJECT_HOMEPAGE_URL@
 Version: @PROJECT_VERSION@
-Requires: @pc_req_public@
 Requires.private: @pc_req_private@
```
Comment (eli-schwartz) [inline libteec/libteec.pc.in:16]: AFAICT `pc_req_private` is also not being used anywhere yet?

**Review by eli-schwartz — CHANGES_REQUESTED** [review]

File: libteec/libteec.pc.in (hunk 1 above)
Comment (nhorman) [reply libteec/libteec.pc.in]: removed

**Comment by jforissier** [comment]
@nhorman please consolidate. It seems to me there should be 3 commits in the end, in that order:
1. cmake: introduce join_paths()
2. cmake: honor CMAKE_INSTALL_LIBDIR/CMAKE_INSTALL_INCLUDEDIR
3. pkgfconfig: remove duplicate flags and unused variables

**Comment by eaaltonen** [comment]
@nhorman, @eli-schwartz: can you clarify your use case a bit? Are you setting some `CMAKE_INSTALL_LIBDIR` locations or?
I'd like to understand why the default
```
exec_prefix=${prefix}
includedir=${prefix}/include
libdir=${exec_prefix}/lib
```
pattern from https://people.freedesktop.org/~dbn/pkg-config-guide.html is not sufficient.

**Comment by nhorman** [comment]
@eaaltonen its for exactly the reason that @eli-schwartz noted above and in PR #334 - because using that pattern fails to honor the cmake macros CMAKE_INSTALL_LIBDIR and CMAKE_INSTALL_INCLUDEDIR.  There is no reason those paths will necessarily be set to the same base location, or end in lib | include for that matter.  Nominally they will, but its not a guarantee.

Also, the freedesktop defined pattern is really more of a guideline anyway, often the default pattern is insufficient for multiarch environment (for instance most distros will set libdir to /usr/lib64 on aarch64/x86-64 platforms, so ${prefix}/lib isn't correct in those cases.

**Comment by eli-schwartz** [comment]
It's not even a default or a guideline -- it's an example.

Examples, by their very definition, are not intended to be treated specially and used verbatim. That example, in particular, if used verbatim, would mean hardcoding `prefix=/usr/local`and NOT filling it in with the value of CMAKE_INSTALL_PREFIX.

**Pull request closed without merging**